python/pyarrow/_compute.pyx

```diff
@@ -1333,6 +1333,43 @@ class IndexOptions(_IndexOptions):
         self._set_options(value)
 
 
+cdef class _MapArrayLookupOptions(FunctionOptions):
+    _occurrence_map = {
+        "ALL": CMapArrayLookupOccurrence_ALL,
+        "FIRST": CMapArrayLookupOccurrence_FIRST,
+        "LAST": CMapArrayLookupOccurrence_LAST,
+    }
+
+    def _set_options(self, scalar, occurrence):
+        try:
+            self.wrapped.reset(
+                new CMapArrayLookupOptions(
+                    pyarrow_unwrap_scalar(scalar),
+                    self._occurrence_map[occurrence]
+                )
+            )
+        except KeyError:
+            _raise_invalid_function_option(occurrence,
+                                           "Should either be FIRST, LAST or ALL")
+
+
+class MapArrayLookupOptions(_MapArrayLookupOptions):
+    """
+    Options for the `map_array_lookup` function.
+
+    Parameters
+    ----------
+    query_key : Scalar
+        The key to search for.
+    occurrence : str
+        The occurrence(s) to return from the MapArray
+        Accepted values are "FIRST", "LAST", "ALL".
```

Comment:
nit, we usually take lowercase options strings, e.g. see the round modes: https://github.com/apache/arrow/blob/10d108d6f0fae2a5265116ff23fe7a0623c2dc2f/python/pyarrow/_compute.pyx#L793-L814